src/views/Notifications/Settings/index.tsx

```diff
@@ -85,6 +96,18 @@ const NotificationSettings = () => {
       ...dirtyInputs
     };
 
+    // if datesChangedNotificationType is not changed by user, but datesChanged is changed, then do the following logic
+    if (!changes.datesChangedNotificationType) {
+      if (changes.datesChanged?.length) {
+        // If Dates Changed notification is subscribed, then manually set datesChangedNotificationType to ALL_MODELS
+        changes.datesChangedNotificationType =
+          DatesChangedNotificationType.ALL_MODELS;
+      } else {
+        // If Dates Changed notification is unsubscribed, set datesChangedNotificationType to null
+        changes.datesChangedNotificationType = null;
+      }
```

Comment:
`datesChangedNotificationType` would already be null, no need to set it again here